### PR TITLE
Contacts 0.04: Added call function to contacts app, Refactoring

### PR DESCRIPTION
**Pull request by devsnd**
- Refactored contacts app
 - Added function to trigger calling a contact on the phone (Requires GadgetBridge with "Allow Intent" permission)

### CODE REVIEW

File: apps/contacts/contacts.app.js [hunk 1]
```diff
@@ -2,108 +2,102 @@
...
+function callNumber (number) {
+  Bluetooth.println(JSON.stringify({
+    t:"intent",
+    target:"activity",
+    action:"android.intent.action.CALL",
+    flags:["FLAG_ACTIVITY_NEW_TASK"],
+    categories:["android.intent.category.DEFAULT"],
+    data: 'tel:' + number,
+  }))
+
```
Comment (nxdefiant) [inline apps/contacts/contacts.app.js:20]: did you check what happens when not connected (Gadgetbridge not installed or similar)?
+1 for refactoring

Comment (devsnd) [reply apps/contacts/contacts.app.js:20]: I only tested in the emulator and it does nothing then. You can always send some data over the wire if the watch is connected via Bluetooth, but I don't know how I could detect if the connected device is an Android and if so, if GadgetBridge is used :man_shrugging:

**Comment by pavelmachek** [comment]
Thanks for doing this. Patches look ok on a quick check. And yes, if someone created a nicer icon, that would be great, too :-).

File: apps/contacts/contacts.app.js (hunk 1 above)
Comment (bobrippling) [reply apps/contacts/contacts.app.js:20]: Yeah, if there's no connected device it'll print and the print gets ignored - I believe printing to some devices/consoles hangs but `Bluetooth` is ok

**Comment by bobrippling** [comment]
Nice! LGTM - tested and looks good too

**Pull request merged**